src/solver_nonlin_nox/4C_solver_nonlin_nox_interface_required_base.hpp

```diff
@@ -20,8 +20,36 @@ namespace NOX
   {
     namespace Interface
     {
-      class RequiredBase : public ::NOX::Epetra::Interface::Required
+      class RequiredBase
       {
+       public:
+        enum FillType
+        {
+          //! The exact residual (F) is being calculated.
+          Residual,
+          //! The Jacobian matrix is being estimated.
+          Jac,
+          //! The preconditioner matrix is being estimated.
+          Prec,
+          //! The fill context is from a FD approximation (includes FDC)
+          FD_Res,
+          //! The fill context is from a MF approximation
+          MF_Res,
+          //! The fill context is from a MF computeJacobian() approximation
+          MF_Jac,
```

Comment:
I am not sure, if everybody is familiar with the abbreviations "FD" and "MF". You could spell them out.

Reply:
@mayrmt I can adjust the documentation. I would keep the enum itself for now as this is a direct copy from `NOX::Epetra`. I can't say if things still work properly if we change them.